core/DataAccess/Model.php

```diff
@@ -149,6 +156,45 @@ public function updateArchiveAsInvalidated($archiveTable, $idSites, $datesByPeri
         return Db::query($sql, $bind);
     }
 
+    /**
+     * @param string $archiveTable Prefixed table name
+     * @param int[] $idSites
+     * @param Segment $segment
+     * @param array $datesByPeriodType
+     * @return \Zend_Db_Statement
+     * @throws Exception
+     */
+    public function updateAllRangeArchivesAsInvalidated($archiveTable, $idSites, $ranges, Segment $segment = null)
+    {
+        $idSites = array_map('intval', $idSites);
+
+        $bind = array();
+
+        $dateConditions = array();
+
+        foreach ($ranges as $range) {
+            // Ranges in the DB match if their date2 is after the start of the search range and date2 is before the end
+            $dateConditions[] = "(date2 >= ? AND date1 <= ?)";
```

Comment:
@katebutler not sure if I'm understanding this correctly.

I have here this query generated:
```sql
UPDATE piwik_archive_numeric_2012_04 SET value = 4 WHERE name LIKE 'done%'
                   AND idsite IN (1)
                   AND (period = 5 AND ((date2 >= ? AND date1 <= ?)))
```

and the binds are `['2019-02-03', '2019-02-04']`. Does `date1` and `date2` maybe need to be switched? The comment mentions twice `date2` so that's also not quite clear. I called it like this:

```
./console core:invalidate-report-data --dates=2019-02-03,2019-02-04 --periods=range
```

Then I would expect date ranges like these to be updated:
```
2019-01-04,2019-03-03
2019-02-04,2019-03-03
2019-01-04,2019-02-03
2019-01-04,2019-02-04
```

but with this query I don't think this works? I might be seeing it wrong though. Or maybe we want it to behave differently? @mattab ?